src/Symfony/Bridge/Twig/Extension/HttpFoundationExtension.php

```diff
@@ -24,13 +25,35 @@
  */
 class HttpFoundationExtension extends AbstractExtension
 {
-    private $requestStack;
-    private $requestContext;
+    private $urlHelper;
 
-    public function __construct(RequestStack $requestStack, RequestContext $requestContext = null)
+    /**
+     * @param UrlHelper $urlHelper
+     */
+    public function __construct($urlHelper)
     {
-        $this->requestStack = $requestStack;
-        $this->requestContext = $requestContext;
+        if ($urlHelper instanceof UrlHelper) {
+            $this->urlHelper = $urlHelper;
+
+            return;
+        }
+
+        if (!$urlHelper instanceof RequestStack) {
+            throw new \TypeError(sprintf('The first argument must be an instance of "%s" or an instance of "%s".', UrlHelper::class, RequestStack::class));
+        }
+
+        @trigger_error(sprintf('Passing an instance of "%s" as the first constructor argument is deprecated since Symfony 4.3, pass an instance of "%s" instead.', RequestStack::class, UrlHelper::class), E_USER_DEPRECATED);
```

Comment:
The message should mention the class that contains the constructor (like `as the first constructor argument of "%s" is`).